IBAnimatable/AnimationType.swift

```diff
@@ -4,60 +4,97 @@
 //
 
 import Foundation
+import UIKit
 
 /**
  Predefined Animation Type
  */
-public enum AnimationType: String {
-  case slideInLeft
-  case slideInRight
-  case slideInDown
-  case slideInUp
-  case slideOutLeft
-  case slideOutRight
-  case slideOutDown
-  case slideOutUp
-  case squeezeInLeft
-  case squeezeInRight
-  case squeezeInDown
-  case squeezeInUp
-  case squeezeOutLeft
-  case squeezeOutRight
-  case squeezeOutDown
-  case squeezeOutUp
-  case fadeIn
-  case fadeOut
-  case fadeOutIn
-  case fadeInOut
-  case fadeInLeft
-  case fadeInRight
-  case fadeInDown
-  case fadeInUp
-  case fadeOutLeft
-  case fadeOutRight
-  case fadeOutDown
-  case fadeOutUp
-  case squeezeFadeInLeft
-  case squeezeFadeInRight
-  case squeezeFadeInDown
-  case squeezeFadeInUp
-  case squeezeFadeOutLeft
-  case squeezeFadeOutRight
-  case squeezeFadeOutDown
-  case squeezeFadeOutUp
-  case zoomIn
-  case zoomOut
+public enum AnimationType {
+  case slide(way: Way, direction: Direction)
+  case squeeze(way: Way, direction: Direction) // miss squeeze
+  case slideFade(way: Way, direction: Direction)
+  case fade(way: FadeWay)
+  case squeezeFade(way: Way, direction: Direction)
+  case zoom(way: Way)
   case shake
   case pop
-  case flipX
-  case flipY
+  case flip(along: Axis)
   case morph
-  case squeeze
   case flash
   case wobble
   case swing
-  case rotate
-  case rotateCCW
-  case moveTo
-  case moveBy
+  case rotate(direction: RotationDirection)
+  case moveTo(x: Double, y: Double)
+  case moveBy(x: Double, y: Double)
+  case none
+  
+  public enum FadeWay: String {
+    case `in`, out, inOut = "inout", outIn = "outin"
+  }
+  public enum Way: String {
+    case out, `in`
+  }
+  public enum Axis: String {
+    case x, y
+  }
+  public enum Direction: String {
+    case left, right, up, down
+    
+    func isVertical() -> Bool {
+      return self == .down || self == .up
+    }
+  }
+  public enum RotationDirection: String {
+    case cw, ccw
+  }
+}
+
+extension AnimationType : IBEnum {
+  
+  public init(string: String?) {
+    guard let string = string else {
+      self = .none
+      return
+    }
+    let nameAndParames = MaskType.extractNameAndParams(from: string.lowercased())
```

Comment:
Cool, just an idea.

Jake

Sent from my iPhone

> On 21 Jul 2016, at 8:30 AM, Jason Ak <notifications@github.com> wrote:
> 
> In IBAnimatable/AnimationType.swift:
> 
> > +      return self == .down || self == .up
> > +    }
> > +  }
> > +  public enum RotationDirection: String {
> > +    case cw, ccw
> > +  }
> > +}
> > +
> > +extension AnimationType : IBEnum {
> > +  
> > +  public init(string: String?) {
> > +    guard let string = string else {
> > +      self = .none
> > +      return
> > +    }
> > +    let nameAndParames = MaskType.extractNameAndParams(from: string.lowercased())
> I don't found how to simply get the Type of self, since self is an enum, 
> I will Replace it with AnimationType.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>